### ISSUE TITLE
[d3d9] Prince of Persia (2008): Post-processing broken - lighting, bloom, more

### ISSUE THREAD

**Issue by KonKode** [doitsujin/dxvk#1537]
This game's "post-processing" setting is completely bonkers in DXVK. Disabling post-processing leads to no obvious bugs. The game works great in WineD3D.

**- GPU:** RX 570
**- Driver:** Mesa 20.0.2
**- Wine version:** lutris-5.4
**- DXVK version:** 1.6, 1.5, and D9VK 0.40.1

**- Log file:** [Prince of Persia_d3d9.log](https://github.com/doitsujin/dxvk/files/4397955/Prince.of.Persia_d3d9.log)
**- Trace:** [Google Drive link](https://drive.google.com/file/d/1JlBg4-NTizbZ88wI1vAPCr7OViB_wWjt/view)

Lighting becomes blown out and static light sources appear "stuck" on the screen as if it were part of a HUD. There's also a good amount of texture dithering and Z-fighting in DXVK. Some examples below, compared against WineD3D.

![](https://i.imgur.com/OsG8xil.jpg)
![](https://i.imgur.com/PaoOJak.jpg)

This momentary flash becomes a wall of artifacts in DXVK. Similar artifacts can occur on a smaller scale when panning the camera pans changes the lighting.
![](https://i.imgur.com/WxVjFv5.png)
![](https://i.imgur.com/61j4hpW.jpg)

My trace file loads into this scene. All this red stuff stays in place even with camera and character movement. This is the "HUD" effect I mentioned.
![](https://i.imgur.com/mmSkieS.jpg)
![](https://i.imgur.com/huUwgxj.jpg)

More static detail lighting at the very start of the game. It persists through this whole section of play in the canyon, until a cutscene.
![](https://i.imgur.com/DWhDZuX.jpg)
![](https://i.imgur.com/lKEuiS2.jpg)

ProtonDB reports look good because the game defaults to NO post-processing. It has to be enabled in a settings app buried in a subfolder, or in the registry.

Thanks for reading this. I'm on Discord if you need further help with the game.

**Comment by KonKode** [comment]
I've confirmed this isn't an ACO issue.

I've also tried the following DLLs in winetricks after looking at my WINEDEBUG log:

d3dcompiler_47
d3dx9_39
d3dxof
d3dx9


Here's the log's loaded DLLs
 
[wineDebug_POP08.log](https://github.com/doitsujin/dxvk/files/4399531/wineDebug_POP08.log)

**Comment by WinterSnowfall** [comment]
> This game's "post-processing" setting is completely bonkers in DXVK.

I hope you don't mind me asking, but which "post-processing" setting are you referring to? I'm unable to find anything similar in the game launcher's graphics tab.

**Comment by KonKode** [comment]
> I hope you don't mind me asking, but which "post-processing" setting are you referring to? I'm unable to find anything similar in the game launcher's graphics tab.

The launcher only has a slider with quality presets, but these represent changes to individual graphics settings in the registry.

There's a **PostEffects** value which is set to 1 on Medium and 2 on High. These errors occur on both 1 and 2, but are more severe on 2 because more effects are enabled.

Setting the value to 0 results in none of these errors appearing.

![pop08Reg](https://user-images.githubusercontent.com/44456824/78110886-bfb2ac80-73c9-11ea-921d-e86625d2ccb7.png)

**Comment by WinterSnowfall** [comment]
> There's a **PostEffects** value which is set to 1 on Medium and 2 on High. These errors occur on both 1 and 2, but are more severe on 2 because more effects are enabled.

I dug a little after I asked and found the registry entries, so I assumed as much, but thanks for confirming.

That being said, I have the GOG version of the game which defaults to the High preset in my case (**PostEffects 2**).

The game looks fine on my GTX 1070, so I'm guessing it's an AMD only issue. The only thing I noticed is that it doesn't fit the screen properly at 1080p, which is probably a game bug. I played it for about half an hour and got to most, if not all, of the scenes you've captured above without finding any issues.

![Prince_Of_Persia](https://user-images.githubusercontent.com/6306593/78112373-ce6b8480-7406-11ea-9b7c-b2a728f4bf5b.jpg)

**Comment by CME42** [comment]
I can only repro this with radv and there is this validation error:
```
VUID-VkSubpassDescriptionDepthStencilResolve-stencilResolveMode-03184(ERROR / SPEC): msgNum: 0 - vkCreateRenderPass2(): Subpass 0 includes a VkSubpassDescriptionDepthStencilResolve structure with invalid stencilResolveMode=2. The Vulkan spec states: The value of stencilResolveMode must be one of the bits set in VkPhysicalDeviceDepthStencilResolveProperties::supportedStencilResolveModes or VK_RESOLVE_MODE_NONE (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkSubpassDescriptionDepthStencilResolve-stencilResolveMode-03184)
    Objects: 1
        [0] 0, type: 0, name: NULL
```

**Comment by Joshua-Ashton** [comment]
Resolved by #1548
